dapps/marketplace/src/pages/create-listing/_RequireShipping.js

```diff
@@ -0,0 +1,212 @@
+import React, { useCallback, useState } from 'react'
+
+import { fbt } from 'fbt-runtime'
+
+import { useMutation } from 'react-apollo'
+
+import EnableMessagingMutation from 'mutations/EnableMessaging'
+import withMessagingStatus from 'hoc/withMessagingStatus'
+
+import MobileModal from 'components/MobileModal'
+
+const EnableEncryptionModal = ({ onEnabled, onClose }) => {
```

Comment:
@sparrowDom Micah mentioned that one of your PR was to redirect users to onboarding if messaging is not enabled, when clicking on Purchase. So, users would have messaging enabled when they reach this page.

Reply:
From my experience this modal popped up to me when I was in a seller role. I think we require messaging onboarding from the buyers but not also from the sellers

Reply:
Sorry, I might have misunderstood your comment. For sellers, they will also go through onboarding flow (also enable messaging). But this modal appears only if you don't have the keys in your localStorage (happens when you clear localStorage or using a different browser/app). 

For buyers, we don't show this or any similar modal. Instead we redirect to onboarding (the profile and email steps would be skipped if he already has an identity).

Reply:
Yup you are right, sellers also go through normal on-boarding - confusion caused by me. 

What I am thinking is when a buyer clicks on "Purchase" we check that he has completed identity + has valid messaging keys in local storage. If not we direct him to onboarding where he needs to finish only the missing steps. 

It would make sense to me that we do the same for the sellers. And not only for shippable listings (as is current state), but when they are trying to create any kind of listing.

cc @micahalcorn